addons/explosives/functions/fnc_dialPhone.sqf

```diff
@@ -28,22 +26,35 @@ private _arr = [];
 for "_i" from 1 to _ran do {
     _arr = _arr + ['.','..','...',''];
 };
+
+private _explosive = [_code] call FUNC(getSpeedDialExplosive);
+
+private _explosiveType = typeOf (_explosive select 0);
+private _ringtone = GVAR(iedRingtonesHash) getOrDefault [_explosiveType,getArray (configFile >> "CfgAmmo" >> _explosiveType >> QGVAR(ringtones)),true];
+if (_ringtone isEqualTo []) then { _ringtone = [[QPATHTOF(Data\Audio\Cellphone_Ring.wss), 0.75]] };
+//support random ringtones
+_ringtone = selectRandom _ringtone;
+_ringtone params ["_ringtonePath", "_ringtoneLenght"];
+
+//Calculate the ringtone duration before detonation with slight randomness
+private _ringtoneDuration = round(count _arr - (_ringtoneLenght + (random 1.5)- 0.75) / 0.25) max 4;
```

Comment:
`_ringtoneLenght + (random 1.5)- 0.75` seems to be too specific to the existing ringtone. Why not just use `random _ringtoneLength` instead?

Or even better: Why not use the config attribute to add randomness on line 24?

Reply:
I think `random _ringtoneLength` might have issues with longer ringtones. 4s ringtone could blow up after 2s of the ringtone end.


> Or even better: Why not use the config attribute to add randomness on line 24?
not sure what you mean? Do you mean that add new config for `(ceil(random 8)) + 1`, so instead of this this random 9s calling time, make it config entry with min1 check?

Reply:
I added both to config